### ISSUE TITLE
Expo-auth-session Expo-crypto getRandomValues - undefined is not a function

### ISSUE THREAD

**Issue by astaninivan1** [expo/expo#22539]
### Minimal reproducible example

Use new AuthSession.AuthRequest()

### Summary

In the product assembly for ios, sometimes an error "undefined is not a function" appears, I managed to find out that it is related to the call "Crypto.getRandomBytes". The problem is floating and is reproduced every other time when trying to login. In the dev assembly everything works correctly

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.3.1
      Shell: 5.1.16 - /usr/local/bin/bash
    Binaries:
      Node: 18.5.0 - ~/.nvm/versions/node/v18.5.0/bin/node
      Yarn: 3.5.0 - ~/.nvm/versions/node/v18.5.0/bin/yarn
      npm: 8.12.1 - ~/.nvm/versions/node/v18.5.0/bin/npm
    Managers:
      CocoaPods: 1.12.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2022.1 AI-221.6008.13.2211.9619390
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      @expo/metro-config: * => 0.7.1 
      @expo/webpack-config: * => 18.0.4 
      expo: ~48.0.15 => 48.0.6 
      react: 18.2.0 => 18.2.0 
      react-dom: * => 18.2.0 
      react-native: 0.71.7 => 0.71.8 
      react-native-web: * => 0.18.12 
    npmGlobalPackages:
      eas-cli: 3.9.1
    Expo Workflow: bare

**Comment by jer-sen** [comment]
I faced the same issue

```
node_modules/base64-js/index.js in getLens at 31:29
    throw new Error('Invalid string. Length must be a multiple of 4')
  }
  // Trim off extra bytes after placeholder bytes are found
  // See: https://github.com/beatgammit/base64-js/issues/42
  var validLen = b64.indexOf('=') ############### here! ##############
  if (validLen === -1) validLen = len
  var placeHoldersLen = validLen === len
    ? 0

        : 4 - (validLen % 4)

node_modules/base64-js/index.js in toByteArray at 55:21
  return ((validLen + placeHoldersLen) * 3 / 4) - placeHoldersLen
}
function toByteArray (b64) {
  var tmp
  var lens = getLens(b64)
  var validLen = lens[0] ############### here! ##############
  var placeHoldersLen = lens[1]
  var arr = new Arr(_byteLength(b64, validLen, placeHoldersLen))

node_modules/expo-crypto/build/Crypto.js in getRandomBytes at 38:27
    if (ExpoCrypto.getRandomBytes) {
        return ExpoCrypto.getRandomBytes(validByteCount);
    }
    else if (ExpoCrypto.getRandomBase64String) {
        const base64 = ExpoCrypto.getRandomBase64String(validByteCount);
        return toByteArray(base64); ############### here! ##############
    }
    else {
        throw new UnavailabilityError('expo-crypto', 'getRandomBytes');
    }
}
```

Architecture | arm64
Battery Level | 94%
Boot Time | 2024-01-06T14:07:38Z(5 days before this event)
Charging | False
Family | iOS
Free memory | 152.2 MiB
Free storage | 5.6 GiB
Memory Size | 3.6 GiB
Model | iPhone 12 mini (D52gAP)
Model Id | D52gAP
Orientation | portrait
Screen Height Pixels | 812
Screen Resolution | 375x812
Screen Width Pixels | 375
Simulator | False
Taille du stockage | 59.5 GiB
Usable Memory | 2.4 GiB

Kernel Version | Darwin Kernel Version 21.6.0: Thu Aug 11 21:46:14 PDT 2022; root:xnu-8020.241.2~1/RELEASE_ARM64_T8101
Name | iOS
Rooted | no
Version | 15.7(19H12)


expo | 49.0.21
expo-crypto | 12.4.1

**Comment by linaia** [comment]
Please reopen : 

expo-crypto Crypto.js

```
else if (ExpoCrypto.getRandomBase64String) {
        const base64 = ExpoCrypto.getRandomBase64String(validByteCount);
        return toByteArray(base64);
    }
```
ExpoCrypto.getRandomBase64String returns base64 that toByteArray is not able to handle ?
toByteArray -> getLens 

```
function getLens (b64) {
  var len = b64.length

  if (len % 4 > 0) {
    throw new Error('Invalid string. Length must be a multiple of 4')
  }

  // Trim off extra bytes after placeholder bytes are found
  // See: https://github.com/beatgammit/base64-js/issues/42
  var validLen = b64.indexOf('=')                        =======>>>>>>>>> fails here as if b64 was not a string (indexOf is not a function) ?
````

**Comment by Charles-Johnson** [comment]
We are observing this occasionally in production using Expo SDK 48, `expo-auth-session@4.03`,  `expo-crypto@12.2.2`

**Comment by Charles-Johnson** [comment]
@linaia I think this was fixed in v12.5.0 in this merge request: https://github.com/expo/expo/pull/24716/files
